pkg/plugins/file/source.go

```diff
@@ -15,49 +15,67 @@
 package file
 
 import (
-	"bytes"
 	"context"
 	"fmt"
 	"io"
 	"os"
 	"strconv"
-	"time"
 
 	"github.com/conduitio/conduit/pkg/foundation/cerrors"
-	"github.com/conduitio/conduit/pkg/plugins"
-	"github.com/conduitio/conduit/pkg/record"
+	"github.com/conduitio/conduit/pkg/plugin/sdk"
 	"github.com/nxadm/tail"
 )
 
 // Source connector
 type Source struct {
-	Tail   *tail.Tail
-	Config map[string]string
+	sdk.UnimplementedSource
 
-	lastPosition record.Position
-	nextOffset   int64
-	nextPosition record.Position
+	tail   *tail.Tail
+	config map[string]string
 }
 
-func (c *Source) Ack(ctx context.Context, position record.Position) error {
-	return nil // no ack needed
+func NewSource() sdk.Source {
+	return &Source{}
 }
 
-func (c *Source) Open(ctx context.Context, config plugins.Config) error {
-	err := c.Validate(config)
+func (s *Source) Configure(ctx context.Context, m map[string]string) error {
+	err := s.validateConfig(m)
 	if err != nil {
 		return err
 	}
-	c.Config = config.Settings
+	s.config = m
 	return nil
 }
 
-func (c *Source) seek(p record.Position) error {
-	if bytes.Equal(c.lastPosition, p) && c.Tail != nil {
-		// we are at the required position
-		return nil
+func (s *Source) Open(ctx context.Context, position sdk.Position) error {
+	return s.seek(position)
+}
+
+func (s *Source) Read(ctx context.Context) (sdk.Record, error) {
+	select {
+	case line := <-s.tail.Lines:
+		return sdk.Record{
+			Position:  sdk.Position(strconv.FormatInt(line.SeekInfo.Offset, 10)),
+			CreatedAt: line.Time,
+			Payload:   sdk.RawData(line.Text),
+		}, nil
+	case <-ctx.Done():
```

Comment:
not needed anymore :D

Reply:
Not really, it is still needed here! We are not using `sdk.ErrBackoffRetry` in this plugin, instead, we are blocking the call until we get a new line _or_ the context gets canceled.